types/hapi/definitions/request/request-events.d.ts

```diff
@@ -26,6 +27,8 @@ export interface RequestEvents extends Podium {
      * * 'disconnect' - emitted when a request errors or aborts unexpectedly.
      * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestevents)
      */
+    on(criteria: "peek", listener: PeekListener): void;
+    on(criteria: "finish" | "disconnect", listener: () => {}): void;
```

Comment:
This implied the listener has to return an object, I don't think that's right.

Reply:
Darn!  Yes good catch!